### PR TITLE
Check the paths of all files before downloading them

### PR DESCRIPTION
**Pull request by hkethi002**
Generates all the paths for files to be downloaded and makes sure there are no conflicts.

### CODE REVIEW

File: export_bids.py [hunk 1]
```diff
@@ -119,10 +166,17 @@ def download_bids_dir(fw, project_id, outdir, src_data=False, dry_run=False):
...
+        'project':{},
+        'session':{},
+        'acquisition':{}
+    }
+
     # Get project
     project = fw.get_project(project_id)
 
-    logger.info('Downloading project files')
+    logger.info('Proccessing project files')
```
Comment (ehlertjd) [inline export_bids.py]: Just a small typo (Processing)

File: export_bids.py [hunk 2]
```diff
@@ -134,30 +188,19 @@ def download_bids_dir(fw, project_id, outdir, src_data=False, dry_run=False):
...
-        # Download the file
-        logger.info('Downloading project file: {0}'.format(f['name']))
 
         # For dry run, don't actually download
-        if dry_run:
-            logger.info('  to {0}'.format(path))
-            continue
+        if path in filepath_downloads['project']:
+            logger.error('Multiple files with {} as path'.format(path))
+            sys.exit(1)
```
Comment (ehlertjd) [inline export_bids.py]: I would print the source filename `f['name']` as part of the error to give the user more context. (Applies to each instance)

**Comment by ehlertjd** [comment]
LGTM

**Comment by hkethi002** [comment]
Because the user wouldn't know which file is going to be the file that is downloaded last, it doesn't seem like a great idea to allow for them to go ahead and download it

**Pull request merged**